### PR TITLE
refactor: fix unparm, gocritic, and errorlint lint errors

### PR DESCRIPTION
**Pull request by AlisonB319**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: cmd/starter/start.go [hunk 1]
```diff
@@ -68,8 +73,9 @@ func startHelper(cmd *cobra.Command, args []string) {
 
 	select {
 	case <-ctx.Done():
-		return
-	case err := <-errorPipeline:
-		log.Fatal(ctx, err.Error())
+		return nil
+	case errPipeline := <-errorPipeline:
```
Comment (cameronmeissner) [inline cmd/starter/start.go:77]: nit: errPipeline -> err?

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -164,12 +164,12 @@ func (t *TemplateGenerator) getSingleLine(textFilename string, profile interface
 	// use go templates to process the text filename
 	templ := template.New("customdata template").Option("missingkey=zero").Funcs(funcMap)
 	if _, err = templ.New(textFilename).Parse(string(b)); err != nil {
-		return "", fmt.Errorf("error parsing file %s: %v", textFilename, err)
+		return "", fmt.Errorf("error parsing file %s: %w", textFilename, err)
```
Comment (cameronmeissner) [inline pkg/agent/baker.go:167]: I guess this is what the linter likes in terms of error formatting?

I've seen a lot of different formatting directives used for error printing in different spots (%s, %v, %w, %q, etc.)

File: pkg/agent/datamodel/versions.go [hunk 3]
```diff
@@ -373,12 +373,14 @@ func getAllKubernetesWindowsSupportedVersionsMap() map[string]bool {
...
 // GetSupportedVersions get supported version list for a certain orchestrator.
-func GetSupportedVersions(orchType string, isUpdate, hasWindows bool) (versions []string, defaultVersion string) {
+func GetSupportedVersions(orchType string, isUpdate, hasWindows bool) ([]string, string) {
+	versions := []string{}
+	defaultVersion := ""
 	switch orchType {
 	case Kubernetes:
 		return GetAllSupportedKubernetesVersions(isUpdate, hasWindows), GetDefaultKubernetesVersion(hasWindows)
 	default:
-		return nil, ""
```
Comment (cameronmeissner) [inline pkg/agent/datamodel/versions.go:381]: the linter was mad at this?

Comment (AlisonB319) [reply pkg/agent/datamodel/versions.go:381]: Yes it wanted a named return I think

**Comment by AlisonB319** [comment]
This PR has some rebase comments. Will be making a new PR that has all of the commits but are still separated by the same commit messages.

**Pull request closed without merging**